frontend/packages/knative-plugin/src/components/add/event-sources/YAMLEditorSection.tsx

```diff
@@ -0,0 +1,15 @@
+import * as React from 'react';
+import YAMLEditorField from '@console/shared/src/components/formik-fields/YAMLEditorField';
+import FormSection from '@console/dev-console/src/components/import/section/FormSection';
+import { useField } from 'formik';
+
+const YAMLEditorSection: React.FC = () => {
+  const [field] = useField('type');
```

Comment:
It should get the type as a prop and this yaml editor section should act as a field component and take care of loading and updating the `yamlData` of your form state based on the prop `type`.

Reply:
updated `YAMLEditorSection` to take prop `type` and accordingly update `yamlData`